parsec/core/cli/utils.py

```diff
@@ -46,50 +47,61 @@ def wrapper(config_dir, *args, **kwargs):
     return wrapper
 
 
-def _unslug(val):
-    parts = val.split(":")
-    if len(parts) == 1:
-        return (None, DeviceID(val), val)
-    elif len(parts) == 2:
-        raw_org, raw_device_id = parts
-        return (OrganizationID(raw_org), DeviceID(raw_device_id), val)
-    else:
-        raise ValueError("Must follow format `[<organization>:]<user_id>@<device_name>`")
+def format_available_devices(devices: List[AvailableDevice]) -> str:
+    # Try to shorten the slughash to make it easier to work with
+    slughashes = [d.slughash for d in devices]
+    for slughash_len in range(3, 64):
+        if len({h[:slughash_len] for h in slughashes}) == len(slughashes):
+            break
+
+    out = []
+    for device in devices:
+        display_slughash = click.style(device.slughash[:slughash_len], fg="yellow")
+        out.append(
+            f"{display_slughash} - {device.organization_id}: {device.user_display} @ {device.device_display}"
+        )
+    return "\n".join(out)
 
 
 def core_config_and_device_options(fn):
     @core_config_options
-    @click.option("--device", "-D", type=_unslug, required=True)
+    @click.option(
+        "--device",
+        "-D",
+        required=True,
+        help="Device to use designed by it ID, see `list_devices` command to get the available IDs",
+    )
     @click.option("--password", "-P")
     @wraps(fn)
     def wrapper(**kwargs):
         config = kwargs["config"]
         password = kwargs["password"]
+        device_slughash = kwargs.pop("device")
+
+        all_available_devices = list_available_devices(config.config_dir)
+        devices = []
+        for device in all_available_devices:
+            if device.slughash.startswith(device_slughash):
+                devices.append(device)
 
-        organization_id, device_id, slugname = kwargs["device"]
-        devices = [
-            (o, d, t, kf)
```

Comment:
Bye `o`, `d`, `t`, and `kf`, we won't miss you~ (especially you `kf`)

Reply:
![image](https://user-images.githubusercontent.com/3187637/84644670-c13b3d80-aeff-11ea-852e-2ede9fcd2091.png)